src/Components/Server/src/Circuits/CircuitHandler.cs

```diff
@@ -72,4 +72,12 @@ public abstract class CircuitHandler
     /// <param name="cancellationToken">The <see cref="CancellationToken"/>.</param>
     /// <returns><see cref="Task"/> that represents the asynchronous execution operation.</returns>
     public virtual Task OnCircuitClosedAsync(Circuit circuit, CancellationToken cancellationToken) => Task.CompletedTask;
+
+    /// <summary>
+    /// Invoked when inbound activity on the circuit causes an asynchronous task to be dispatched on the server.
+    /// </summary>
+    /// <param name="circuit">The <see cref="Circuit"/>.</param>
+    /// <param name="next">The next handler to invoke.</param>
+    /// <returns></returns>
+    public virtual Task HandleInboundActivityAsync(Circuit circuit, Func<Task> next) => next();
```

Comment:
Does it make sense to separate this into its own interface? I would prefer if we can avoid deeply nested calls if all we are doing is threading through the handlers and in most cases they do nothing.

Reply:
The default implementation doesn't really favor discovery a lot, and it imposes a cost even when we noop

Reply:
I see the concern about no-ops. Do you think anyone is going to have so many circuit handlers that the perf impact would to be significant? Even though this new method has a different shape than the others, I think it aligns with the overall purpose of `CircuitHandler`.

That said, I'm fine if we separate it out. Maybe we could call it something like `ICircuitInboundEventHandler`? And do you think each implementor would be its own service (like `CircuitHandler`), or should we have some other way to register handlers? Maybe something like an `AddCircuitEventHandlers(Action<CircuitEventHandlerOptions>)` method on `IServerSideBlazorBuilder`?